CBinding/Parser/ClangProjectSymbolDatabase.cs

```diff
@@ -0,0 +1,442 @@
+using System;
+using ClangSharp;
+using System.Collections.Generic;
+using GLib;
+using System.Runtime.InteropServices;
+using System.Linq;
+using CBinding.Navigation;
+
+namespace CBinding.Parser
+{	
+	/// <summary>
+	/// Symbol database belonging to a project
+	/// </summary>
+	public class ClangProjectSymbolDatabase {
```

Comment:
Longer term this needs to be serializable so it doesn't have to be rebuilt every time. It could probably be built using the clang_index* functions.